### PR TITLE
Rename project to libgitops

### PR DESCRIPTION
**Pull request by twelho**
Rename all internal references to `gitops-toolkit` to the new name `libgitops`, this change covers the code as well as any documentation.

Also includes some minor improvements to the `Makefile` to ease the vendoring and tidying process.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,14 +1,17 @@
...
 ## Getting Help
 
-If you have any questions about, feedback for or problems with `gitops-toolkit`:
+If you have any questions about, feedback for or problems with `libgitops`:
 
-- Invite yourself to the <a href="https://slack.weave.works/" target="_blank">Weave Users Slack</a>.
-- Ask a question on the [#ignite](https://weave-community.slack.com/messages/general/) slack channel.
-- [File an issue](https://github.com/weaveworks/gitops-toolkit/issues/new).
+- Invite yourself to the [Weave Users Slack](https://slack.weave.works/).
+- Ask a question on the [#ignite](https://weave-community.slack.com/messages/general/) Slack channel.
```
Comment (luxas) [inline README.md:11]: this links to the general channel right? it should be. then also please mark it as #general

File: README.md [hunk 2]
```diff
@@ -1,14 +1,17 @@
...
+- [File an issue](https://github.com/weaveworks/libgitops/issues/new).
 
 Your feedback is always welcome!
+
+## Notes
+This project was formerly called `gitops-toolkit`, but has now been given a more descriptive name.
```
Comment (luxas) [inline README.md:17]: Also add a comment and link to the "real" GitOps Toolkit at https://github.com/fluxcd/toolkit

File: README.md (hunk 1 above)
Comment (twelho) [reply README.md:11]: Should we be directing to #ignite or #general for questions related to this?

File: README.md (hunk 2 above)
Comment (twelho) [reply README.md:17]: :+1:

File: README.md (hunk 1 above)
Comment (twelho) [reply README.md]: I'll go with #general, since this has no Ignite-specific functionality.

**Pull request merged**